src/veriff/veriff.service.ts

```diff
@@ -0,0 +1,171 @@
+      throw new UnauthorizedException('Invalid Veriff signature.');
+    }
+
+    try {
+      const vc = await this.credentialFactory.create<VeriffPayload>(jwt);
+
+      const did = vc.getId();
+
+      if (!did) {
+        this.logger.warn('VC is missing DID, cannot proceed.');
+        throw new BadRequestException('Verifiable Credential is missing DID.');
+      }
+
+      // Check pepSanctionMatches
+      if (data.verification.decision === 'approved') {
+        try {
+          let pepSanctionMatches: WatchlistScreeningResult | undefined;
+          if (ENABLE_PEP_CHECK) {
+            pepSanctionMatches =
+              await this.veriffWatchlistService.getWatchlistScreening(
+                payload.sessionId,
+              );
+            this.logger.debug('Watchlist result:', pepSanctionMatches);
+          }
+        } catch (e) {
+          this.logger.warn('Failed to fetch watchlist screening:', e.message);
+          throw new InternalServerErrorException(
+            `Veriff webhook processing failed: ${e.message}`,
+          );
+        }
+
+        const issuer = await getTonomyOpsIssuer();
+
+        const person = data.verification.person;
+
+        const firstName = getFieldValue(person, 'firstName');
+        const lastName = getFieldValue(person, 'lastName');
+        const birthDate = getFieldValue(person, 'dateOfBirth');
+        const nationality = getFieldValue(person, 'nationality');
+
+        if (!firstName || !birthDate || !nationality) {
+          throw new Error('Missing required personal data.');
+        }
+
+        const signedFirstNameVc = await signVerifiableCredential(
+          'FirstNameCredential',
+          { firstName },
+          issuer,
+          did,
+        );
+
+        const signedLastNameVc = await signVerifiableCredential(
+          'LastNameCredential',
+          { lastName },
+          issuer,
+          did,
+        );
+
+        const signedBirthDateVc = await signVerifiableCredential(
+          'BirthDateCredential',
+          { birthDate },
+          issuer,
+          did,
+        );
+
+        const signedNationalityVc = await signVerifiableCredential(
+          'NationalityCredential',
+          { nationality },
+          issuer,
+          did,
+        );
+
+        const signedVc = await signVerifiableCredential(
+          'VeriffCredential',
+          { verification: payload },
+          issuer,
+          did,
+        );
+
+        const mewPayload = JSON.stringify({
+          kyc: signedVc,
+          firstName: signedFirstNameVc,
+          lastName: signedLastNameVc,
+          birthDate: signedBirthDateVc,
+          nationality: signedNationalityVc,
+        });
+
+        this.communicationGateway.sendVeriffVerificationToDid(did, mewPayload);
+      } else {
+        this.logger.debug(
+          'Verification decision is not approved, skipping response data.',
+        );
+        throw new BadRequestException(
```

Comment:
not sure that this is vhat we agreed on. should discuss this. It's OK to merge for now, but we MUST discuss before release

Reply:
decided that for now we will need to send decision to user, the `signedVc` even if not approved, so that the front end can move on to the nextscreen